### PR TITLE
DH examples from https://tools.ietf.org/rfc/rfc5114.txt

### PR DESCRIPTION
**Pull request by public**
A.{4,5,6,7,8} are omitted as they are ECDH examples.

https://tools.ietf.org/rfc/rfc5114.txt

```
   The test data in this appendix is a protocol-independent subset of
   the test data in [EX80056A].  In the test data for the three modular
   exponentiation groups, we use the following notation:

      xA: The secret key of party A

      yA: The public key of party A

      xB: The secret key of party B

      yB: The public key of party B

      Z: The shared secret that results from the Diffie-Hellman
         computation
```

### CODE REVIEW

**Comment by public** [comment]
I'll be using these instead of the NIST vectors in #1721

**Comment by reaperhulk** [comment]
Needs documentation in `test-vectors.rst`

**Comment by alex** [comment]
I think historically we've reformatted IETF vectors?

**Comment by reaperhulk** [comment]
We have if we needed to for our current parsers to handle them. @public do you have to write a new vector loader for this?

**Comment by public** [comment]
I can convert them to something our existing loaders can cope with.

**Comment by reaperhulk** [comment]
Yeah let's do that.

**Comment by public** [comment]
Converted for use with the KASVS DH loader.

File: vectors/cryptography_vectors/asymmetric/DH/RFC5114.txt [hunk 1]
```diff
@@ -0,0 +1,44 @@
+[A.1.  1024-bit MODP Group with 160-bit Prime Order Subgroup]
+P = B10B8F96A080E01DDE92DE5EAE5D54EC52C99FBCFB06A3C69A6A9DCA52D23B616073E28675A23D189838EF1E2EE652C013ECB4AEA906112324975C3CD49B83BFACCBDD7D90C4BD7098488E9C219A73724EFFD6FAE5644738FAA31A4FF55BCCC0A151AF5F0DC8B4BD45BF37DF365C1A65E68CFDA76D4DA708DF1FB2BC2E4A4371
+Q = F518AA8781A8DF278ABA4E7D64B7CB9D49462353
+G = A4D1CBD5C3FD34126765A442EFB99905F8104DD258AC507FD6406CFF14266D31266FEA1E5C41564B777E690F5504F213160217B4B01B886A5E91547F9E2749F4D7FBD7D3B9A92EE1909D0D2263F80A76A6A24C087A091F531DBF0A0169B6A28AD662A4D18E73AFA32D779D5918
```
Comment (reaperhulk) [inline vectors/cryptography_vectors/asymmetric/DH/RFC5114.txt]: This doesn't look right? Shouldn't this be `A4D1CBD5C3FD34126765A442EFB99905F8104DD258AC507FD6406CFF14266D31266FEA1E5C41564B777E690F5504F213160217B4B01B886A5E91547F9E2749F4D7FBD7D3B9A92EE1909D0D2263F80A76A6A24C087A091F531DBF0A0169B6A28AD662A4D18E73AFA32D779D5918D08BC8858F4DCEF97C2A24855E6EEB22B3B2E5`?

Comment (public) [inline vectors/cryptography_vectors/asymmetric/DH/RFC5114.txt]: Just testing you ;)

Comment (public) [inline vectors/cryptography_vectors/asymmetric/DH/RFC5114.txt]: Fixed.

**Comment by chelseawinfree** [comment]
Verified these as true compared to the rfc listed.

**Pull request merged**